src/test/txindex_tests.cpp

```diff
@@ -70,9 +70,6 @@ BOOST_FIXTURE_TEST_CASE(txindex_initial_sync, TestChain100Setup)
     // shutdown sequence (c.f. Shutdown() in init.cpp)
     txindex.Stop();
 
-    threadGroup.interrupt_all();
-    threadGroup.join_all();
-
```

Comment:
Not sure why this should be safe to remove. The scheduler needs to be stopped *before* this test ends, so the `interrupt_all` should be replaced by a `scheduler.stop()`. I think that otherwise you reintroduce bug #15410

Reply:
Thanks. I was just now looking at src/test/checkqueue_tests.cpp which has similar code.

Reply:
@jonatack In transaction_tests, that's a locally declared `threadGroup`, and it's a `BasicTestingSetup` too, so the `TestingSetup` destructor wouldn't execute, and even if it did, wouldn't clean up that `threadGroup`.

@MarcoFalke Ah, you're quite right. Have undeleted the code and added a comment.

Reply:
@ajtowns right, makes sense. Thanks for the explanation.

Reply:
Calling `stop()` doesn't actually cause the thread to stop, you need to wait for it to complete. Could do that by looping and waiting for `AreThreadsServicingQueue()` to be false, but that still leaves the possibility you continue on and destruct the object in between `--nThreadsServicingQueue;` and `newTaskScheduled.notify_one();` at the end of `serviceQueue`. Could drop the `interrupt_all` but if some other thread in the group needed an interrupt before being joined, that'd cause a hang too... So until we change something else, this doesn't seem redundant to me?

Reply:
Ups, correct.